contracts/Zap.sol

```diff
@@ -615,16 +622,16 @@ contract Zap is Ownable, ReentrancyGuard {
             deadline
         );
 
-        if (transferResidual) {
-            // returning residue in tokenA, if any
-            if (amountADesired - amountA > 0) {
-                TransferHelper.safeTransfer(tokenA, msg.sender, (amountADesired - amountA));
-            }
+        uint256 residualTokenA = amountADesired - amountA;
+        // returning residue in tokenA, if any
+        if (residualTokenA > 0 && residualTokenA >= transferResidualMinTokenA) {
```

Comment:
doing residualTokenA > transferResidualMinTokenA would allow avoid comparing against 0

Reply:
I thought about it, it's such a small thing but in the case of `residualTokenA == transferResidualMinTokenA` and assuming it's also `>0` we would still be donating those tokens to the Zap contract when we shouldn't 🤔

Reply:
Updated anyway: chances of `residualTokenA == transferResidualMinTokenA` are smaller than winning the Euromillion jackpot, so gas savings for everyone instead!